packages/rum-core/src/domain/rumEventsCollection/view/trackInteractionToNextPaint.ts

```diff
@@ -0,0 +1,142 @@
+ */
+function processEntry(newInteraction: RumEventTiming | RumFirstInputTiming) {
+  const interaction = find(
+    longestInteractions,
+    (interaction) => newInteraction.interactionId === interaction.interactionId
+  )
+  const minLongestInteraction = longestInteractions[longestInteractions.length - 1]
+
+  if (interaction) {
+    if (newInteraction.duration > interaction.duration) {
+      const index = longestInteractions.indexOf(interaction)
+      longestInteractions[index] = newInteraction
+      sortAndTrimLongestInteractions()
+    }
+  } else if (
+    longestInteractions.length < MAX_INTERACTION_ENTRIES ||
+    newInteraction.duration > minLongestInteraction.duration
+  ) {
+    longestInteractions.push(newInteraction)
+    sortAndTrimLongestInteractions()
+  }
+}
+
+/**
+ * Compute the p98 longest interaction.
+ * For better performance the computation is based on 10 longest interactions and the interaction count of the current view.
+ */
+function estimateP98LongestInteraction(getInteractionCount: () => number) {
+  const interactionIndex = Math.min(longestInteractions.length - 1, Math.floor(getInteractionCount() / 50))
+  return longestInteractions[interactionIndex]
+}
+
+function sortAndTrimLongestInteractions() {
+  longestInteractions.sort((a, b) => b.duration - a.duration).splice(MAX_INTERACTION_ENTRIES)
+}
+
+function matchInteraction(candidate: RumEventTiming | RumFirstInputTiming) {
+  return longestInteractions.some(
+    (interaction) => candidate.duration === interaction.duration && candidate.startTime === interaction.startTime
+  )
+}
+
+let previousInteractionCount: number
+function trackInteractionCount(viewLoadingType: ViewLoadingType) {
+  if ('interactionCount' in performance) {
+    previousInteractionCount = (window as BrowserWindow).performance.interactionCount!
+    return {
+      getInteractionCount: () =>
+        viewLoadingType === ViewLoadingType.INITIAL_LOAD
+          ? (window as BrowserWindow).performance.interactionCount!
+          : (window as BrowserWindow).performance.interactionCount! - previousInteractionCount,
+      stop: noop,
+    }
+  }
+
+  let interactionCount = 0
+  let minKnownInteractionId = Infinity
+  let maxKnownInteractionId = 0
+
+  const observer = new (window as BrowserWindow).PerformanceObserver((entries: PerformanceObserverEntryList) => {
+    entries.getEntries().forEach((e) => {
+      const entry = e as unknown as RumEventTiming
+      if (entry.interactionId) {
+        minKnownInteractionId = Math.min(minKnownInteractionId, entry.interactionId)
+        maxKnownInteractionId = Math.max(maxKnownInteractionId, entry.interactionId)
+
+        interactionCount = maxKnownInteractionId ? (maxKnownInteractionId - minKnownInteractionId) / 7 + 1 : 0
+      }
+    })
+  })
+  observer.observe({ type: 'event', buffered: viewLoadingType === ViewLoadingType.INITIAL_LOAD, durationThreshold: 0 })
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏why `durationThreshold: 0`? 
from [mdn](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceObserver/observe#syntax), the min should be 16 🤔

Reply:
Yes a durationThreshold is capped to 16 but like setTimeout(.., 0) is capped to 4ms I think setting 0 better conveys the idea that we want all the ente entries.

Reply:
In any case I isolated the interaction count polyfill from Google from the rest of the logic to ease the reading :)